akka-cluster/src/main/scala/akka/cluster/ClusterRemoteWatcher.scala

```diff
@@ -90,14 +94,22 @@ private[cluster] class ClusterRemoteWatcher(
   def memberRemoved(m: Member, previousStatus: MemberStatus): Unit =
     if (m.address != selfAddress) {
       clusterNodes -= m.address
-      // The reason we don't quarantine gracefully removed members (leaving) is that
-      // Cluster Singleton need to exchange TakeOver/HandOver messages.
+
       if (previousStatus == MemberStatus.Down) {
         quarantine(m.address, Some(m.uniqueAddress.uid), s"Cluster member removed, previous status [$previousStatus]")
+      } else if (arteryEnabled) {
+        // don't quarantine gracefully removed members (leaving) directly,
+        // give Cluster Singleton some time to exchange TakeOver/HandOver messages.
+        import context.dispatcher
+        context.system.scheduler.scheduleOnce(cluster.downingProvider.downRemovalMargin, self, DelayedQuarantine(m, previousStatus))
```

Comment:
Not sure this is a good idea, but if the cluster singleton doesn't have a specific removalMargin configured, this is what is used as removalMargin in ClusterSingletonManager.

I'm guessing we can't really depend on the singleton manager setting here.